File: app/src/main/java/com/example/noteappmvvm/Repository/NoteRepository.java
package com.example.noteappmvvm.Repository;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.example.noteappmvvm.Dao.NotesDao;
import com.example.noteappmvvm.Database.NotesDatabase;
import com.example.noteappmvvm.Model.Notes;

import java.util.List;

public class NoteRepository {

    public NotesDao notesDao;
    public LiveData<List<Notes>> getallNotes;
    public LiveData<List<Notes>> hightoLow;
    public LiveData<List<Notes>> lowtoHigh;


    public  NoteRepository(Application application) {
        NotesDatabase database = NotesDatabase.getDatabaseInstance(application);
        notesDao = database.notesDao();
        getallNotes = notesDao.getallNotes();
        hightoLow = notesDao.hightoLow();
        lowtoHigh = notesDao.lowtoHigh();

    }

    public void insertNotes(Notes notes) {
        notesDao.insertNotes(notes);
    }

    public void deleteNotes(int id) {
        notesDao.deleteNotes(id);
    }

    public void updateNotes(Notes notes) {
        notesDao.updateNotes(notes);
    }
}

File: app/src/main/java/com/example/noteappmvvm/Repository/NoteRepository.kt
package com.example.noteappmvvm.Repository

import android.app.Application
import androidx.lifecycle.LiveData
import com.example.noteappmvvm.Dao.NotesDao
import com.example.noteappmvvm.Database.NotesDatabase.Companion.getDatabaseInstance
import com.example.noteappmvvm.Model.Notes

class NoteRepository(application: Application?) {
    private var notesDao: NotesDao?
    @JvmField
    var getallNotes: LiveData<List<Notes?>?>?
    @JvmField
    var hightoLow: LiveData<List<Notes?>?>?
    @JvmField
    var lowtoHigh: LiveData<List<Notes?>?>?
    fun insertNotes(notes: Notes?) {
        notesDao!!.insertNotes(notes)
    }

    fun deleteNotes(id: Int) {
        notesDao!!.deleteNotes(id)
    }

    fun updateNotes(notes: Notes?) {
        notesDao!!.updateNotes(notes)
    }

    init {
        val database = getDatabaseInstance(application!!)
        notesDao = database!!.notesDao()
        getallNotes = notesDao!!.getallNotes()
        hightoLow = notesDao!!.hightoLow()
        lowtoHigh = notesDao!!.lowtoHigh()
    }
}